bin/propolis-server/src/lib/vm/ensure.rs

```diff
@@ -180,6 +180,12 @@ impl<'a> VmEnsureNotStarted<'a> {
             state: MachineInitializerState::default(),
         };
 
+        // If we are publishing metrics to oximeter, then create a kstat-sampler
+        // now. This is used to track the vCPUs today, but will soon be used to
+        // track instance datalink stats as well. It can be provided to
+        // `initialize_network_devices()` at that time.
+        let kstat_sampler = init.create_kstat_sampler();
```

Comment:
Is it worthwhile to create the sampler if the `producer_registry` in the `options` is `None`? (In particular, does the `kstat_sampler`'s memory usage grow over time if it never gets collected from? We'll have a producer in production Propolises, of course, but it's sometimes useful to be able to run long-running tests with a freestanding propolis-server binary.)

Reply:
The `create_kstat_sampler()` method returns an `Option<KstatSampler>`. That's `None` if either the `producer_registry` is `None`, or if we just fail to create the thing. I'll rename this variable to make that more clear!

That obviates your second question, about memory consumption, but the answer is "no". The `KstatSampler` requires one to choose a limited buffer size, used in a ringbufffer to maintain uncollected samples. The memory consumption is bounded, even if it's never polled for data.

Reply:
That makes sense--thanks for the explanation!